moml/models/mgnn/hmgnn.py

```
@@ -0,0 +1,408 @@
+    """
+    Multi-head attention across S scales (0 = finest).
+    Optionally preceded by an EdgeNNConv message pass on each fine→coarse pair.
+    """
+
+    def __init__(self, n_scales: int, hidden_dim: int, n_heads: int = 4, edge_dim: int = 0):
+        super().__init__()
+        assert hidden_dim % n_heads == 0, "hidden_dim must be divisible by n_heads"
+        self.S, self.h, self.d_k = n_scales, n_heads, hidden_dim // n_heads
+        self.scale = nn.Parameter(torch.tensor(math.sqrt(self.d_k)))  # learnable temperature
+
+        make_proj = lambda: nn.ModuleList(nn.Linear(hidden_dim, hidden_dim) for _ in range(n_scales))
```

Ruff (0.11.9)

Do not assign a `lambda` expression, use a `def`

Rewrite `make_proj` as a `def`

(E731)